src/coreclr/src/jit/emit.cpp

```diff
@@ -4637,7 +4645,7 @@ unsigned emitter::emitEndCodeGen(Compiler* comp,
     }
 
     UNATIVE_OFFSET roDataAlignmentDelta = 0;
-    if (emitConsDsc.dsdOffs)
+    if (emitConsDsc.dsdOffs && (emitConsDsc.alignment == TARGET_POINTER_SIZE))
```

Comment:
This would probably be better suited to having a `CORJIT_ALLOCMEM_FLG_RODATA_8BYTE_ALIGN`, but I figured it was out of scope.
Given we track the alignment request of all constants, we can determine if aligning to 8 is even necessary.